src/uu/seq/src/extendedbigdecimal.rs

```diff
@@ -0,0 +1,249 @@
+        match self {
+            ExtendedBigDecimal::BigDecimal(x) => {
+                let (n, p) = x.as_bigint_and_exponent();
+                match p {
+                    0 => ExtendedBigDecimal::BigDecimal(BigDecimal::new(n * 10, 1)).fmt(f),
+                    _ => x.fmt(f),
+                }
+            }
+            ExtendedBigDecimal::Infinity => f32::INFINITY.fmt(f),
+            ExtendedBigDecimal::MinusInfinity => f32::NEG_INFINITY.fmt(f),
+            ExtendedBigDecimal::MinusZero => {
+                // FIXME In Rust version 1.53.0 and later, the display
+                // of floats was updated to allow displaying negative
+                // zero. See
+                // https://github.com/rust-lang/rust/pull/78618. Currently,
+                // this just formats "0.0".
+                (0.0f32).fmt(f)
+            }
+            ExtendedBigDecimal::Nan => "nan".fmt(f),
+        }
+    }
+}
+
+impl Zero for ExtendedBigDecimal {
+    fn zero() -> Self {
+        ExtendedBigDecimal::BigDecimal(BigDecimal::zero())
+    }
+    fn is_zero(&self) -> bool {
+        match self {
+            Self::BigDecimal(n) => n.is_zero(),
+            Self::MinusZero => true,
+            _ => false,
+        }
+    }
+}
+
+impl Add for ExtendedBigDecimal {
+    type Output = Self;
+
+    fn add(self, other: Self) -> Self {
+        match (self, other) {
+            (Self::BigDecimal(m), Self::BigDecimal(n)) => Self::BigDecimal(m.add(n)),
+            (Self::BigDecimal(_), Self::MinusInfinity) => Self::MinusInfinity,
+            (Self::BigDecimal(_), Self::Infinity) => Self::Infinity,
+            (Self::BigDecimal(_), Self::Nan) => Self::Nan,
+            (Self::BigDecimal(m), Self::MinusZero) => Self::BigDecimal(m),
+            (Self::Infinity, Self::BigDecimal(_)) => Self::Infinity,
+            (Self::Infinity, Self::Infinity) => Self::Infinity,
+            (Self::Infinity, Self::MinusZero) => Self::Infinity,
+            (Self::Infinity, Self::MinusInfinity) => Self::Nan,
+            (Self::Infinity, Self::Nan) => Self::Nan,
+            (Self::MinusInfinity, Self::BigDecimal(_)) => Self::MinusInfinity,
+            (Self::MinusInfinity, Self::MinusInfinity) => Self::MinusInfinity,
+            (Self::MinusInfinity, Self::MinusZero) => Self::MinusInfinity,
+            (Self::MinusInfinity, Self::Infinity) => Self::Nan,
+            (Self::MinusInfinity, Self::Nan) => Self::Nan,
+            (Self::Nan, _) => Self::Nan,
+            (Self::MinusZero, other) => other,
+        }
+    }
+}
+
+impl PartialEq for ExtendedBigDecimal {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::BigDecimal(m), Self::BigDecimal(n)) => m.eq(n),
+            (Self::BigDecimal(_), Self::MinusInfinity) => false,
+            (Self::BigDecimal(_), Self::Infinity) => false,
+            (Self::BigDecimal(_), Self::Nan) => false,
+            (Self::BigDecimal(_), Self::MinusZero) => false,
```

Comment:
I'll make that change.

Reply:
Oh actually, this is intentionally `false`.

The reason I am representing `MinusZero` as a distinct concept at all is because if `-0.0` is given in the input as the starting number, it must be displayed as `-0.0` in the output. Because `-0.0` gets rendered as `"0.0"` in older versions of Rust, we need to know whether the value is `ExtendedBigDecimal::MinusZero` or the value is `ExtendedBigDecimal::BigDecimal(BigDecimal::zero())`. The equality comparison happens when we are deciding how to render the current value in `write_value_float()`:
```
if *value == ExtendedBigDecimal::MinusZero && is_first_iteration {
```
https://github.com/uutils/coreutils/pull/2698/files#diff-8ef9575cb5c2b35e15751308bf63d6dbf3ac217a383790f9355bc5e13ab2fdb9R231

If the `BigDecimal::zero()` and `MinusZero` were to compare equal, then the `write_value_float()` would write `"-0.0"` when it should be writing `"0.0"`. For example,
```
$ ./target/debug/coreutils seq -w 0.0 1
-0.0
1.0
```
We don't want that, so we need to ensure that negative zero and positive zero are considered different things.

When the minimum Rust version becomes large enough, we may be able to eliminate much of the special handling code for negative zero.